cpp/include/cuspatial/experimental/detail/linestring_distance.cuh

```diff
@@ -241,15 +241,13 @@ void pairwise_linestring_distance(OffsetIterator linestring1_offsets_first,
                               typename std::iterator_traits<OutputIt>::value_type>(),
     "Inputs and output must have floating point value type.");
 
-  static_assert(detail::is_same<T,
-                                typename std::iterator_traits<Cart2dItB>::value_type::value_type,
-                                typename std::iterator_traits<OutputIt>::value_type>(),
+  static_assert(detail::is_same<T, typename std::iterator_traits<OutputIt>::value_type>(),
                 "Inputs and output must have the same value type.");
 
   static_assert(detail::is_same<cartesian_2d<T>,
                                 typename std::iterator_traits<Cart2dItA>::value_type,
                                 typename std::iterator_traits<Cart2dItB>::value_type>(),
-                "Inputs must be cuspatial::cartesian_2d");
+                "Input types mismatches or input types are not cuspatial::cartesian_2d");
```

Comment:
I think the former error message was clearer and sufficient, since the mismatch of underlying float type is handled by the previous assertion.

```suggestion
                "Input types must be cuspatial::cartesian_2d");
```

Reply:
I think the former assertion only checks whether the input types are of floating types, it's still only possible to mix double with floats? 
https://godbolt.org/z/T1Y5TfMa1

We can avoid this when `is_same_floating_point` is introduced.

Reply:
But you have checks that they are all the same already, so this error message does not need to mention that. And there is a grammar error (types mismatches --> types mismatch)

Reply:
Grammar error aside... I think the changes to line 155 made it it's only checking if linestring 1 is of the same `value_type` to the output. It doesn't check if the linestring 1 and linestring 2 has the same `value_type`.  Consider if the user has the following input:

```
Cart2dItA: iterator to cartesian_2d<float> type
Cart2dItB: iterator to cartesian_2d<double> type
OutputIt: iterator to float type
```

It passes `is_floating_point` check, because all are floating point type. It passes first `is_same` check, because `Cart2dItA::value_type` and `OutputIt` are both `float`. It fails at the second `is_same` check, because `cartesian_2d<float> != cartesian_2d<double>`. But your suggestion would make this line to throw compile error
`Input types must be cuspatial::cartesian_2d`, which is confusing because the user does supply a `cartesian_2d` type.

The error would make sense for the following input:
```
struct MockType {
    using value_type = float;
};

Cart2dItA: iterator to MockType
Cart2dItB: iterator to MockType
OutputIt: iterator to float type
```

After `is_convertible_to` is introduced in #579 , we should use that here. Since `cartesian_2d<float>` shouldn't be convertible to `cartesian_2d<double>`, we should have the same type safety here.

Reply:
Good point. Still, I think the message can be simpler. 

```suggestion
                "Input types must be cuspatial::cartesian_2d with the same value type");
```